joplin/api/schema.py

```diff
@@ -307,6 +307,19 @@ def resolve_departments(cls, instance, info):
         return instance.departments()
 
 
+class RelatedEventPageResolver(graphene.Interface):
+    events = graphene.List(lambda: EventPageNode)
+
+    @classmethod
+    def resolve_events(cls, instance, info):
+        events = []
+        event_relationships = EventPageRelatedPage.objects.filter(page__id=instance.pk).order_by('event__date')
+        for page in event_relationships:
+            if not page.event.canceled:
+                events.append(page.event)
+        return events
```

Comment:
events are returned in date order and only if they are not canceled.

Reply:
and the filtering is happening all the way down at the [component level in janis](https://github.com/cityofaustin/janis/pull/754/files#diff-e5c49d73ef8d69841d449238bd8b2807R9), beautiful